terraform/generate-batch.tf

```diff
@@ -0,0 +1,195 @@
+# Compute Environment Launch Template
+resource "aws_launch_template" "aws_batch_ce_lt" {
+  name = "${var.prefix}-batch-compute-environment"
+  block_device_mappings {
+    device_name = "/dev/xvda"
+    ebs {
+      volume_size           = 30
+      delete_on_termination = true
+      encrypted             = true
+      volume_type           = "gp2"
+    }
+  }
+  user_data              = filebase64("user-data")
+  update_default_version = "true"
+}
+
+# MODIS Aqua
+# Compute Environment
+resource "aws_batch_compute_environment" "generate_aqua" {
+  compute_environment_name = "${var.prefix}-aqua"
+  compute_resources {
+    allocation_strategy = "BEST_FIT_PROGRESSIVE"
+    ec2_configuration {
+      image_id_override = data.aws_ssm_parameter.ecs_image_id.value
+      image_type        = "ECS_AL2"
+    }
+    instance_role = aws_iam_instance_profile.ecs_instance_profile.arn
+    instance_type = var.instance_type
+    launch_template {
+      launch_template_id = aws_launch_template.aws_batch_ce_lt.id
+      version            = aws_launch_template.aws_batch_ce_lt.latest_version
+    }
+    max_vcpus          = 32
+    min_vcpus          = 0
+    security_group_ids = data.aws_security_groups.vpc_default_sg.ids
+    subnets            = data.aws_subnets.private_application_subnets.ids
+    type               = "EC2"
+    tags = {
+      "Name" : "${var.prefix}-batch-aqua-node"
+    }
+  }
+  service_role = aws_iam_role.aws_batch_service_role.arn
+  state        = "ENABLED"
+  tags         = local.default_tags
+  type         = "MANAGED"
+
+  provisioner "local-exec" {
```

Comment:
Little concerned about these `local-exec` provisioners on the `aws_batch_compute_environment` resources. This requires that whatever machine is running terraform also has the correct version of the aws command line tool installed and has valid aws credentials configured for the specific profile being used.

Is this just enabling `containerInsights`? Is it worth enabling; what does it do for us? I see you can enable it on `aws_ecs_cluster` resources but it seems we aren't actually using an ecs cluster?

Reply:
Makes total sense! The `local-exec` provisioner enables Container Insights as there is no Terraform parameter to do so. In the past I have found Container Insights helpful to track resource usage and fine tune the container job definitions. AWS Batch does use ECS clusters to run jobs on. However, I think it makes sense to remove it from the Terraform files and I will enable it manually for testing. It sounds like we may not need to gather these metrics in production.

Reply:
Yes I think that approach will be better. Let's us simplify the terraform a bit and we can always enable manually for debugging.